lredis/protocol.lua

```diff
@@ -36,43 +39,81 @@ local function encode_inline(arg)
 	return table.concat(arg, " ", 1, n+1)
 end
 
+-- Encode and send a redis command.
+local function send_command(file, arg)
+	local req, err_code = encode_request(arg)
+	if not req then
+		return nil, err_code
+	end
+	req, err_code = file:write(req)
+	if not req then
+		return nil, err_code
+	end
+	req, err_code = file:flush()
+	if not req then
+		return nil, err_code
+	end
+
+	return true
+end
+
 -- Parse a redis response
 local function read_response(file, new_status, new_error, string_null, array_null)
-	local line = assert(file:read("*l"))
-	assert(line:sub(-1, -1) == "\r", "invalid line ending")
+	local line, err_code = file:read("*l")
+	if not line then
+		return nil, err_code or "EOF reached"
+	elseif line:sub(-1, -1) ~= "\r" then
+		return nil, "invalid line ending"
+	end
+
 	local status, data = line:sub(1, 1), line:sub(2, -2)
 	if status == "+" then
 		return new_status(data)
 	elseif status == "-" then
 		return new_error(data)
 	elseif status == ":" then
-		return assert(tonumber(data, 10), "invalid integer")
+		return tonumber(data, 10), "invalid integer"
```

Comment:
don't return an error message when an error didn't occur

Reply:
Ok, happy to change this.